src/indra_cogex/sources/wikidata/__init__.py

```diff
@@ -0,0 +1,431 @@
+            f"Missing CiteScore data for {missing_citescore_data} journals"
+        )
+
+    def process_data(self, force: bool = False):
+        """Dump data to CSV
+
+        Parameters
+        ----------
+        force: bool
+            If True, force the dump even if the files already exist
+        """
+        if self.publisher_data_path.exists() and \
+                self.journal_data_path.exists() and \
+                self.pub_jour_relations_data_path.exists() and \
+                not force:
+            logger.info("Files already exist, skipping dump.")
+            return
+
+        with gzip.open(self.publisher_data_path, 'wt') as publisher_fh, \
+                gzip.open(self.journal_data_path, 'wt') as journal_fh, \
+                gzip.open(self.pub_jour_relations_data_path, 'wt') as \
+                        relations_fh:
+            publisher_writer = csv.writer(publisher_fh, delimiter='\t')
+            journal_writer = csv.writer(journal_fh, delimiter='\t')
+            relations_writer = csv.writer(relations_fh, delimiter='\t')
+
+            used_isni = set()
+            used_nlm = set()
+            nlm_isni_relations = set()
+            skipped = 0
+            for journal_publisher in self.iter_data():
+                # Skip if no nlm_id or publisher_isni
+                nlm_id = journal_publisher.nlm_id
+                isni = journal_publisher.publisher_isni
+
+                if not isni or not nlm_id:
+                    skipped += 1
+                    continue
+
+                # Save relations
+                # One relations file for journal-publisher relations
+                # One row per relation:
+                # nlm_id, publisher_isni
+                if (nlm_id, isni) not in nlm_isni_relations:
+                    relations_writer.writerow((nlm_id, isni))
+                    nlm_isni_relations.add((nlm_id, isni))
+
+                # Save publishers
+                # One row per Publisher:
+                # publisher_wd_id, publisher_name, publisher_isni
+                if isni not in used_isni:
+                    publisher_writer.writerow([
+                        journal_publisher.publisher_wd_id,
+                        journal_publisher.publisher_name,
+                        isni
+                    ])
+                    used_isni.add(isni)
+
+                # Save journals
+                # One row per Journal:
+                # journal_wd_id, journal_name, issn_list, journal_issn_l,
+                # nlm_id, citescore, category_rank, percentile, category,
+                # citations_2019_22, documents_2019_22, percent_cited_2019_22,
+                # snip, sjr
+                if journal_publisher.nlm_id not in used_nlm:
+                    journal_writer.writerow([
+                        journal_publisher.journal_wd_id,
+                        journal_publisher.journal_name,
+                        json.dumps(journal_publisher.journal_issn_list or []),
```

Comment:
maybe better to pipe delimit or comma delimit?

Reply:
directly semicolon delimit is good for neo4j later as it won't need additional processing

Reply:
Agreed.

Resolved in https://github.com/bgyori/indra_cogex/pull/139/commits/a740f19d444a710cc429cc6679a637d2f4bf5e78